service/modules/lin-reg-module/linreg.py

```diff
@@ -0,0 +1,58 @@
+import mlflow
+import mlflow.sklearn
+import json
+import sys
+import logging
+
+import numpy as np
+import pandas as pd
+
+from burdock.client import get_dataset_client
+from burdock.data.adapters import load_reader, load_metadata, load_dataset
+
+from diffprivlib.mechanisms import Vector
+from diffprivlib.utils import PrivacyLeakWarning, DiffprivlibCompatibilityWarning, warn_unused_args
+from diffprivlib.tools.utils import mean
+from diffprivlib.models import LinearRegression
+
+if __name__ == "__main__":
+    dataset_name = sys.argv[1]
+    budget = float(sys.argv[2])
+    # We expect the next two inputs in the following format: json.dumps([col, names, here]) (e.g. '["a","b"]')
+    x_features = json.loads(sys.argv[3])
+    y_targets = json.loads(sys.argv[4])
+
+    with mlflow.start_run(run_name="diffpriv_linreg"):
+        dataset_document = get_dataset_client().read(dataset_name, budget)
+        dataset = load_dataset(dataset_document)
+        schema = load_metadata(dataset_document)
+
+        # Find X and y values from dataset
+        X = dataset[x_features]
+        y = dataset[y_targets]
+
+        # Find ranges for X and ranges for y
+        table_name = dataset_name + "." + dataset_name
+        x_range_dict = dict([(col, schema[table_name][col].maxval - schema[table_name][col].minval)
+                             for col in x_features])
+        y_range_dict = dict([(col, schema[table_name][col].maxval - schema[table_name][col].minval)
+                             for col in y_targets])
+        x_range = pd.Series(data=x_range_dict)
+        y_range = pd.Series(data=y_range_dict)
+
+        # TODO:
+        #  Currently calculating the data norm instead of using schema-specified
+        #  Eventually, this will need to be changed, as it is bad practice
+        norms = np.linalg.norm(dataset, axis=1)
+        max_norm = np.amax(norms)
+        logging.warning('Currently calculating the data norm instead of using schema-specified value. \
+                        This is bad practice, and will eventually be changed')
+
+        model = LinearRegression(data_norm=max_norm, epsilon=budget, range_X=x_range, range_y=y_range).fit(X, y)
+
+        # Log mlflow attributes for mlflow UI
+        mlflow.log_param("dataset_name", dataset_name)
```

Comment:
the params should be logged at the start of the run. It helps keep more info even when things fail